### DISCUSSION TITLE
issue: append() trigger handleSubmit() and move() trigger append()

### DISCUSSION THREAD

**Discussion by reksamamur** [react-hook-form/react-hook-form#8981] [Q&A] [answered]
### Version Number

7.3.4.2

### Codesandbox/Expo snack

https://codesandbox.io/s/sandbox-react-form-dynamic-form-38hvw3?file=/src/App.tsx

### ****Steps to reproduce****

1. See the example sandbox 
2. Add the first field by clicking “First Field” button
3. Add the parent field by clicking “Add” button and you’ll see in the console it triggers `handleSubmit()` 
    1. the Same thing happens when adding a child or nested field, clicking “Add Nested” will trigger `handleSubmit()` 
4. If you click “Move Up” and “Move Down” it’ll trigger `handleSubmit()` and “Move Down” at the last item will trigger `append()` adding a new field

### **Expected behavior**

`handleSubmit()` should be triggered when clicking submit button, and “Move Down” at the last field should not be triggering `append()` adding a new item

### **What browsers do you see the problem on?**

*No response*

### **Relevant log output**

*No response*

### **Code of Conduct**

- [x]  I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
missing button attribute `type="button"`

**Reply by reksamamur** [reply]
Ok, got it! it's all work now, thanks!

As for why "Move Down" button triggers `append()`, I think it's because `index + 1` in `move(index, index + 1)`. 
So to make it not add a new field or triggers `append()` I add validation to disable the button `index === fields.length`.

https://codesandbox.io/s/sandbox-react-form-dynamic-form-38hvw3?file=/src/App.tsx
